e2e/dractions/disapp.go

```diff
@@ -39,7 +53,7 @@ func EnableProtectionDiscoveredApps(ctx types.TestContext) error {
 	}
 
 	drpc := generateDRPCDiscoveredApps(
-		name, managementNamespace, cluster.Name, drPolicyName, placementName, appname, appNamespace)
+		cluster.Name, drPolicyName, appname, ctx, recipeRef)
```

Comment:
Why so many arguments changed when the only thing we need to do is add recipeRef? It seems that you added unrelated refactoring here. Lets avoid unrelated changes. If we need to do some refactoring to make the needed change easier, make the refactoring in a separate commit with a good commit message explaining the change.

Reply:
Good idea but this change but this should not be mixed with unrelated changes. Do this refactoring in a separate commit before adding the new arguments. Also keep ctx as the first argument like we do in all other functions.